pkg/providers/version/version.go

```diff
@@ -69,32 +70,26 @@ func NewDefaultProvider(kubernetesInterface kubernetes.Interface, cache *cache.C
 }
 
 func (p *DefaultProvider) Get(ctx context.Context) (string, error) {
-	var version string
-	var versionSource string
 	if version, ok := p.cache.Get(kubernetesVersionCacheKey); ok {
 		return version.(string), nil
 	}
-	output, err := p.eksapi.DescribeCluster(ctx, &eks.DescribeClusterInput{
-		Name: lo.ToPtr(options.FromContext(ctx).ClusterName),
-	})
-	if err != nil {
-		if !awserrors.IsAccessDenied(err) {
-			return "", err
-		}
-		output, err := p.kubernetesInterface.Discovery().ServerVersion()
-		if err != nil {
-			return "", fmt.Errorf("getting kubernetes version from the kubernetes API")
-		} else if output != nil {
-			version = fmt.Sprintf("%s.%s", output.Major, strings.TrimSuffix(output.Minor, "+"))
-			versionSource = "Kubernetes API"
-		}
-	} else if lo.FromPtr(output.Cluster.Version) != "" {
-		version = *output.Cluster.Version
-		versionSource = "EKS DescribeCluster"
-	} else {
-		return "", fmt.Errorf("unable to retrieve Kubernetes version from EKS DescribeCluster")
+
+	var version, versionSource string
+	var err error
+
+	if options.FromContext(ctx).EKSControlPlane {
+		version, versionSource, err = p.getEKSVersion(ctx)
```

Comment:
We know the version source if we pulled it from the EKS control plane, so it feels odd that we would return this from our getEKSVersion function